modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_serializer.py

```diff
@@ -256,7 +256,8 @@ def serialize_literal(self, literal):
         dict
             Serialized literal.
         """
-        if literal.val is None:
+        val = literal.val
+        if val is None or isinstance(val, (float, np.float64)) and np.isnan(val):
```

Comment:
probably can just do like this, it should detect all kinds of none values (None, float("nan"), np.nan ...):
```suggestion
        if pandas.isna(val):
```